okapi-core/src/main/java/org/folio/okapi/managers/ProxyService.java

```diff
@@ -254,27 +252,24 @@ private String tenantHeader(ProxyContext pc) {
     String tok = ctx.request().getHeader(XOkapiHeaders.TOKEN);
 
     if (auth != null) {
-      Pattern pattern = Pattern.compile("Bearer\\s+(.+)"); // Grab anything after 'Bearer' and whitespace
-      Matcher matcher = pattern.matcher(auth);
-      if (matcher.find() && matcher.groupCount() > 0) {
-        auth = matcher.group(1);
+      if (auth.startsWith("Bearer")) {
+        auth = auth.substring(6).trim();
+      }
+      if (tok != null && !auth.equals(tok)) {
+        pc.responseError(400, messages.getMessage("10104"));
+        return null;
+      } else {
```

Comment:
Can you drop this `else` to reduce indentation?

Reply:
yep